app/models/assignment.rb

```diff
@@ -600,6 +600,20 @@ def summary_json(user)
       numMarked: self.get_num_marked(user.admin? ? nil : user.id) }
   end
 
+  # Generate a JSON summary of the test results associated with an assignment
+  def summary_test_result_json(user)
+    return {} unless user.admin? || user.ta?
```

Comment:
You can remove this, and the `user` parameter. We do authorization at the controller level, and the default rule, `AssignmentPolicy#manage?` will apply and block requests from students (and, in fact, from graders that don't have permissions to manage assignments).